src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -65,19 +65,22 @@ export class WorkspaceSavedObjectsClientWrapper {
 
     return [permission];
   }
-  private async validateSingleWorkspacePermissions(
-    workspaceId: string | undefined,
+  private async validateSingleObjectPermissions(
+    id: string | undefined,
+    type: string,
     request: OpenSearchDashboardsRequest,
     permissionMode: WorkspacePermissionMode | WorkspacePermissionMode[]
   ) {
-    if (!workspaceId) {
+    // PermissionMode here is an array which is merged by workspace type required permission and other saved object required permission.
```

Comment:
I couldn't get this, could you elaborate a bit?
> PermissionMode here is an array which is merged by workspace type required permission and other saved object required permission.

Reply:
Previously, our validation logic for savedobject was: 
```
if(isWorkspaceTypeSavedObject){
validate(workspaceType, workspaceTypeNeedPermissions)
}else{
validate(type, savedObjectNeedPermissions)
}
```
Now, we can merge workspaceTypeNeedPermissions and savedObjectNeedPermissions and unify savedObject permission code.